src/features/DebugSession.ts

```diff
@@ -114,6 +124,87 @@ export class DebugSessionFeature extends LanguageClientConsumer
+                return undefined;
+            }
+
+
+
+            // Ensure the .NET session stops before the PowerShell session so that the .NET debug session doesnt emit an error about the process unexpectedly terminating
+            const startDebugEvent = vscode.debug.onDidStartDebugSession((dotnetAttachSession) => {
+                // Makes the event one-time
+                // HACK: This seems like you would be calling a method on a variable not assigned yet, but it does work in the flow.
+                // The dispose shorthand demonry for making an event one-time courtesy of: https://github.com/OmniSharp/omnisharp-vscode/blob/b8b07bb12557b4400198895f82a94895cb90c461/test/integrationTests/launchConfiguration.integration.test.ts#L41-L45
+                startDebugEvent.dispose();
+                this.logger.write(`Debugger session detected: ${dotnetAttachSession.name} (${dotnetAttachSession.id})`);
+                if (dotnetAttachSession.configuration.name == dotnetDebuggerConfig.name) {
+                    const stopDebugEvent = vscode.debug.onDidTerminateDebugSession(async (terminatedDebugSession) => {
+                        // Makes the event one-time
+                        stopDebugEvent.dispose();
+
+                        this.logger.write(`Debugger session stopped: ${terminatedDebugSession.name} (${terminatedDebugSession.id})`);
+
+                        if (terminatedDebugSession === session) {
+                            this.logger.write("Terminating dotnet debugger session associated with PowerShell debug session");
+                            await vscode.debug.stopDebugging(dotnetAttachSession);
+                        }
+                    });
+                }
+            });
+
+            // Start a child debug session to attach the dotnet debugger
+            // TODO: Accomodate multi-folder workspaces if the C# code is in a different workspace folder
+            await vscode.debug.startDebugging(undefined, dotnetDebuggerConfig, session);
+            this.logger.write(`Attached dotnet debugger to process ${pid}`);
+        }
+        return this.tempSessionDetails;
+    }
+
+    public getDotnetNamedConfigOrDefault(session: DebugSession, pid: number) {
+        if (session.configuration.dotnetDebuggerConfigName) {
+            const debugConfigs = vscode.workspace.getConfiguration("launch").get<DebugConfiguration[]>("configurations") ?? [];
+
+            return debugConfigs.find((config) => {
+                return config.type === "coreclr"
+                    && config.request === "attach"
+                    && config.name === config.dotnetDebuggerConfigName;
+            });
+        }
+
+        // Default debugger config if none provided
+        // TODO: Type this appropriately from the C# extension?
+        return {
+            name: "Dotnet Debugger: PSIC Temporary Console",
```

Comment:
```suggestion
            name: "Dotnet Debugger: Temporary Debug Terminal",
```

We made a concerted effort away from "PowerShell Integrated Console" as we found the name confusing (to me and to users). The only references that should be left are in actual setting strings because those are...annoying...to change (and migrate). See: https://github.com/PowerShell/vscode-powershell/issues/4053 so the PSIC became "PowerShell Extension Terminal" and I think the temporary debug terminals should be just that, "Temporary Debug Terminal."

Reply:
I just didn't update it after picking the PR back up and I totally agree, thanks!